src/pool/srv_util.c

```diff
@@ -451,3 +451,29 @@ ds_pool_check_failed_replicas(struct pool_map *map, d_rank_list_t *replicas,
 	}
 	return 0;
 }
+
+
+int ds_pool_get_ranks_by_uuid(const uuid_t pool_uuid, int status,
+			      d_rank_list_t *ranks)
```

Comment:
seems better to comment that user need to do map_ranks_init() for succeed call.

Or change the API to be 
int ds_pool_get_ranks_by_uuid(const uuid_t pool_uuid, int status,
			      d_rank_list_t **ranks);
that do map_ranks_init/map_ranks_fini(), + d_rank_list_dup() internally, and caller need to do d_rank_list_free().

Reply:
ah, ok